modules/network/src/main/java/org/apache/ignite/internal/network/serialization/ClassDescriptor.java

```diff
@@ -52,21 +52,28 @@
     @Nullable
     private final ClassDescriptor superClassDescriptor;
 
+    /**
+     * Component type descriptor (only present for arrays).
+     */
+    @Nullable
+    private final ClassDescriptor componentTypeDescriptor;
+
+    private final boolean isPrimitive;
```

Comment:
Looks odd that this fields are declared without newlines and javadocs (as opposed to other fields)

Reply:
They do not need javadocs, the name seems to be sufficient. Some fields need javadocs because they are not so obvious. I think it's a good idea to do this selectively.
As for the newlines, I use them here for grouping fields. In this case, the 4 fields constitute a group, they look as one block, it's intentional.